security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -492,12 +492,15 @@ func receiveThread(con *sdsConnection, reqChannel chan *xdsapi.DiscoveryRequest,
 	for {
 		req, err := con.stream.Recv()
 		if err != nil {
+			con.mutex.RLock()
```

Comment:
Add some comment to explain what error this lock can prevent?

Reply:
added